libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/task/TaskId.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks.task;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Class encapsulating Task identifier
+ */
+@ExperimentalApi
+public class TaskId {
+
+    /**
+     * Identified of the Task
+     */
+    private final String id;
```

Comment:
Why do we have a dedicated class for this at this point?

Reply:
We dont need-need this at this point, but trying to avoid primitive data types references in `Task` so that it keeps room for extending later without major change/breaking stuff